### PR TITLE
Remove support for tags, also remove the tags column from the member table

### PR DESCRIPTION
**Pull request by whirm**
Malicious proof & co. is also gone.

### CODE REVIEW

File: community.py [hunk 1]
```diff
@@ -1956,15 +1943,13 @@ def on_messages(self, messages):
 
         The messages are processed with the following steps:
 
-         1. Messages created by a member in our blacklist are droped.
+         1. Messages that are old or duplicate, based on their distribution policy, are dropped.
```
Comment (NielsZeilemaker) [inline community.py]: lets say replaced or superceded instead of old

File: dispersydatabase.py [hunk 2]
```diff
@@ -380,6 +374,33 @@ def check_database(self, database_version):
...
+-- remove table malicious_proof
+DROP TABLE IF EXISTS malicious_proof;
+-- update database version
+UPDATE option SET value = '18' WHERE key = 'database_version';
+""")
+                self.commit()
+                logger.debug("upgrade database %d -> %d (done)", database_version, 19)
+
+            if database_version < 19:
                 # there is no version 18 yet...
```
Comment (NielsZeilemaker) [inline dispersydatabase.py]: 19

File: member.py [hunk 3]
```diff
@@ -220,50 +213,16 @@ def has_identity(self, community):
...
-            if not tag in self._tags:
-                # the tag isn't there to begin with
-                return False
-            self._tags.remove(tag)
-
-        self._database.execute(u"UPDATE member SET tags = ? WHERE id = ?", (u",".join(sorted(self._tags)), self._database_id))
-        return True
-
+    # TODO(emilon): decide what to do with this
     @property
```
Comment (NielsZeilemaker) [inline member.py]: remove it

File: tests/test_undo.py [hunk 4]
```diff
@@ -191,99 +191,6 @@ def test_node_undo_other(self):
         self._dispersy.get_community(community.cid).unload_community()
 
     @call_on_dispersy_thread
-    def test_self_malicious_undo(self):
```
Comment (NielsZeilemaker) [inline tests/test_undo.py:194]: keep this test, rename it to test_undo_twice

**Comment by whirm** [comment]
This should be ready to be merged

**Comment by whirm** [comment]
retest this please

**Comment by whirm** [comment]
@NielsZeilemaker, @LipuFei This should be ready to be merged

**Pull request merged**